ZIGSIMPlusTests/AppSettingModelTests.swift

```diff
@@ -0,0 +1,74 @@
+    }
+
+    override func tearDown() {
+    }
+
+    func testIsCameraEnabled_WithoutParameter_WhenAllCameraCommandsInactive() {
+        AppSettingModel.shared.isActiveByCommand[.arkit] = false
+        AppSettingModel.shared.isActiveByCommand[.ndi] = false
+        AppSettingModel.shared.isActiveByCommand[.imageDetection] = false
+        XCTAssertFalse(AppSettingModel.shared.isCameraEnabled())
+    }
+    
+    func testIsCameraEnabled_WithoutParameter_WhenArkitActive() {
+        AppSettingModel.shared.isActiveByCommand[.arkit] = true
+        AppSettingModel.shared.isActiveByCommand[.ndi] = false
+        AppSettingModel.shared.isActiveByCommand[.imageDetection] = false
+        XCTAssertTrue(AppSettingModel.shared.isCameraEnabled())
+    }
+
+    func testIsCameraEnabled_ExceptArkit_WhenAllCameraCommandsInactive() {
+        AppSettingModel.shared.isActiveByCommand[.arkit] = false
+        AppSettingModel.shared.isActiveByCommand[.ndi] = false
+        AppSettingModel.shared.isActiveByCommand[.imageDetection] = false
+        XCTAssertFalse(AppSettingModel.shared.isCameraEnabled(except: .arkit))
+    }
+
+    func testIsCameraEnabled_ExceptArkit_WhenNdiActive() {
+        AppSettingModel.shared.isActiveByCommand[.arkit] = false
+        AppSettingModel.shared.isActiveByCommand[.ndi] = true
+        AppSettingModel.shared.isActiveByCommand[.imageDetection] = false
+        XCTAssertTrue(AppSettingModel.shared.isCameraEnabled(except: .arkit))
+    }
+ 
+    func testIsCameraEnabled_ExceptNdi_WhenAllCameraCommandsInactive() {
+        AppSettingModel.shared.isActiveByCommand[.arkit] = false
+        AppSettingModel.shared.isActiveByCommand[.ndi] = false
+        AppSettingModel.shared.isActiveByCommand[.imageDetection] = false
+        XCTAssertFalse(AppSettingModel.shared.isCameraEnabled(except: .ndi))
+    }
+    
+    func testIsCameraEnabled_ExceptNdi_WhenImageDetectionActive() {
+        AppSettingModel.shared.isActiveByCommand[.arkit] = false
+        AppSettingModel.shared.isActiveByCommand[.ndi] = false
+        AppSettingModel.shared.isActiveByCommand[.imageDetection] = true
+        XCTAssertTrue(AppSettingModel.shared.isCameraEnabled(except: .ndi))
+    }
+    
+    func testIsCameraEnabled_ExceptImageDetection_WhenAllCameraCommandsInactive() {
+        AppSettingModel.shared.isActiveByCommand[.arkit] = false
+        AppSettingModel.shared.isActiveByCommand[.ndi] = false
+        AppSettingModel.shared.isActiveByCommand[.imageDetection] = false
+        XCTAssertFalse(AppSettingModel.shared.isCameraEnabled(except: .imageDetection))
+    }
+    
+    func testIsCameraEnabled_ExceptImageDetection_WhenArkitActive() {
+        AppSettingModel.shared.isActiveByCommand[.arkit] = true
+        AppSettingModel.shared.isActiveByCommand[.ndi] = false
+        AppSettingModel.shared.isActiveByCommand[.imageDetection] = false
+        XCTAssertTrue(AppSettingModel.shared.isCameraEnabled(except: .imageDetection))
+    }
```

Comment:
Cool!
FYI, if you wanna cover all cases you can generate assertion like this: 

```swift
// [(arkit, ndi, imageDetection, expected)]
let cases = [
    (false, false, false, (false, false, false, false)),
    (false, false, true, (true, true, true, false)),
    (false, true, false, (true, true, false, false)),
    (true, false, false, (true, false, true, true)),
]

for case in cases {
    let aps = AppSettingModel.shared
    aps.isActiveByCommand[.arkit] = case.1
    aps.isActiveByCommand[.ndi] = case.2
    aps.isActiveByCommand[.imageDetection] = case.3

    let exp = case.4
    XCTAssertEqual(AppSettingModel.shared.isCameraEnabled(), exp.1)
    XCTAssertEqual(AppSettingModel.shared.isCameraEnabled(except: .arkit), exp.2)
    XCTAssertEqual(AppSettingModel.shared.isCameraEnabled(except: .ndi), exp.3)
    XCTAssertEqual(AppSettingModel.shared.isCameraEnabled(except: .imageDetection), exp.4)
}
```

Reply:
I see, using tuples makes it simpler.
Thanks!